public/pages/DetectorResults/containers/AnomalyResultsTable.tsx

```diff
@@ -62,6 +67,85 @@ export function AnomalyResultsTable(props: AnomalyResultsTableProps) {
       ? props.anomalies.filter((anomaly) => anomaly.anomalyGrade > 0)
       : [];
 
+  const handleOpenDiscover = async (startTime: number, endTime: number, item: any) => {
+    try {
+      // calculate time range with 10-minute buffer on each side per customer request
+      const TEN_MINUTES_IN_MS = 10 * 60 * 1000;
+      const startISO = new Date(startTime - TEN_MINUTES_IN_MS).toISOString();
+      const endISO = new Date(endTime + TEN_MINUTES_IN_MS).toISOString();
+      
+      const basePath = `${window.location.origin}${window.location.pathname.split('/app/')[0]}`;
+      
+      const savedObjectsClient = getSavedObjectsClient();
+      
+      const indexPatternTitle = props.detectorIndex.join(',');
+      
+      // try to find an existing index pattern with this title
+      const indexPatternResponse = await savedObjectsClient.find({
+        type: 'index-pattern',
+        fields: ['title'],
+        search: `"${indexPatternTitle}"`,
+        searchFields: ['title'],
+      });
+      
+      let indexPatternId;
+      
+      if (indexPatternResponse.savedObjects.length > 0) {
+        indexPatternId = indexPatternResponse.savedObjects[0].id;
+      } else {
+        // try to create a new index pattern
+        try {
+          const newIndexPattern = await savedObjectsClient.create('index-pattern', {
+            title: indexPatternTitle,
+            timeFieldName: props.detectorTimeField,
+          });
+          
+          indexPatternId = newIndexPattern.id;
+
+          getNotifications().toasts.addSuccess(`Created new index pattern: ${indexPatternTitle}`);
+        } catch (error) {
+          getNotifications().toasts.addDanger(`Failed to create index pattern: ${error.message}`);
+          return;
+        }
+      }
+      
+      // put query params for HC detector
+      let queryParams = '';
+      if (props.isHCDetector && item[ENTITY_VALUE_FIELD]) {
+        const [field, value] = item[ENTITY_VALUE_FIELD].split(': ').map((s: string) => s.trim());
```

Comment:
This does not work for multiple categorical fields.

Reply:
When I clicking host:server_1, service:app_3, this is what I get 

![Screenshot 2025-04-03 at 10 06 02 AM](https://github.com/user-attachments/assets/0f7875d5-3e3d-4365-9c57-6c0b796cf0d2)

Reply:
oops, forgot entity could have multiple values. 

updated the code logic to handle it and here's a screenshot in Discover that selects two entities.

also, in UTs, I'm using two entities values in the mocked anomalies data. 
![Screenshot 2025-04-03 at 14 37 17](https://github.com/user-attachments/assets/985dee35-0253-48df-ae47-d24b645e3b3e)